wcfsetup/install/files/lib/system/background/job/UnfurlUrlBackgroundJob.class.php

```diff
@@ -0,0 +1,240 @@
+            case 2:
+                // 30 minutes
+                return 30 * 60;
+            case 3:
+                // 2 hours
+                return 2 * 60 * 60;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function perform()
+    {
+        $unfurlUrl = new UnfurlUrl($this->urlID);
+
+        try {
+            $unfurlResponse = UnfurlResponse::fetch($unfurlUrl->url);
+
+            if (empty(StringUtil::trim($unfurlResponse->getTitle()))) {
+                $this->save(UnfurlUrl::STATUS_REJECTED);
+
+                return;
+            }
+
+            $title = StringUtil::truncate($unfurlResponse->getTitle(), 255);
+            $description = "";
+            if ($unfurlResponse->getDescription()) {
+                $description = StringUtil::truncate($unfurlResponse->getDescription());
+            }
+
+            if ($unfurlResponse->getImageUrl()) {
+                try {
+                    $image = $this->downloadImage($unfurlResponse->getImage());
+                    $imageData = \getimagesizefromstring($image);
+                    if ($imageData !== false) {
+                        $imageType = $this->validateImage($imageData);
+                        if (!(MODULE_IMAGE_PROXY || IMAGE_ALLOW_EXTERNAL_SOURCE)) {
+                            $imageHash = $this->saveImage($imageData, $image);
+                        } else {
+                            $imageHash = "";
+                        }
+                    } else {
+                        $imageType = UnfurlUrl::IMAGE_NO_IMAGE;
+                    }
+
+                    if ($imageType === UnfurlUrl::IMAGE_NO_IMAGE) {
+                        $imageUrl = $imageHash = "";
+                    } else {
+                        $imageUrl = $unfurlResponse->getImageUrl();
+                    }
+                } catch (UrlInaccessible | DownloadFailed $e) {
+                    $imageType = UnfurlUrl::IMAGE_NO_IMAGE;
+                    $imageUrl = $imageHash = "";
+                }
+            } else {
+                $imageType = UnfurlUrl::IMAGE_NO_IMAGE;
+                $imageUrl = $imageHash = "";
+            }
+
+            $this->save(
+                UnfurlUrl::STATUS_SUCCESSFUL,
+                $title,
+                $description,
+                $imageType,
+                $imageUrl,
+                $imageHash
+            );
+        } catch (UrlInaccessible | ParsingFailed $e) {
+            if (\ENABLE_DEBUG_MODE) {
+                \wcf\functions\exception\logThrowable($e);
+            }
+
+            $this->save(UnfurlUrl::STATUS_REJECTED);
+        }
+    }
+
+    private function downloadImage(Response $imageResponse): string
+    {
+        $image = "";
+        while (!$imageResponse->getBody()->eof()) {
+            $image .= $imageResponse->getBody()->read(8192);
+
+            if ($imageResponse->getBody()->tell() >= UnfurlResponse::MAX_IMAGE_SIZE) {
+                break;
```

Comment:
Thinking about this: A `break;` probably is not super useful here. The image request should simply fail of the response length is too long. Otherwise we'll truncate the image, leading to a broken image.

Thus I suggest to:
```suggestion
                return "";
```

and then removing the `Range` request header. This should likewise be adjusted in ImageProxy.